### PR TITLE
Set NotNull constraint for `this` in SE

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Contributes to #5860

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs [hunk 1]
```diff
@@ -24,6 +24,9 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors
 {
     internal static class References
     {
+        public static ProgramState ProcessThis(SymbolicContext context) =>
+            context.State.SetOperationValue(context.Operation, SymbolicValue.This);     // Implicit and Explicit
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs:28]: Can we have a test for "implicit"?

Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs:28]: The same concerns about `SetOperationValue` dropping other constraints apply here too. See the comments in https://github.com/SonarSource/sonar-dotnet/pull/5884/files#r925797569

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. One general concern about `public static readonly SymbolicValue This = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);` is relevant here too. Therefore I keep it open and don't merge it until that concern is resolved.

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs:28]: Implicit is more difficult to assert, but doable

**Pull request merged**